test/test_transforms_v2_refactored.py

```diff
@@ -2280,3 +2281,59 @@ def resize_my_datapoint():
         _register_kernel_internal(F.resize, MyDatapoint, datapoint_wrapper=False)(resize_my_datapoint)
 
         assert _get_kernel(F.resize, MyDatapoint) is resize_my_datapoint
+
+
+class TestPermuteChannels:
+    _CORRECTNESS_PERMUTATIONS = list(itertools.permutations(range(3)))
+    _DEFAULT_PERMUTATION = _CORRECTNESS_PERMUTATIONS[-1]
+
+    @pytest.mark.parametrize(
+        ("kernel", "make_input"),
+        [
+            (F.permute_channels_image_tensor, make_image_tensor),
+            # FIXME
+            # check_kernel does not support PIL kernel, but it should
+            (F.permute_channels_image_tensor, make_image),
+            (F.permute_channels_video, make_video),
+        ],
+    )
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel(self, kernel, make_input, dtype, device):
+        check_kernel(kernel, make_input(dtype=dtype, device=device), permutation=self._DEFAULT_PERMUTATION)
+
+    @pytest.mark.parametrize(
+        ("kernel", "make_input"),
+        [
+            (F.permute_channels_image_tensor, make_image_tensor),
+            (F.permute_channels_image_pil, make_image_pil),
+            (F.permute_channels_image_tensor, make_image),
+            (F.permute_channels_video, make_video),
+        ],
+    )
+    def test_dispatcher(self, kernel, make_input):
+        check_dispatcher(F.permute_channels, kernel, make_input(), permutation=self._DEFAULT_PERMUTATION)
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.permute_channels_image_tensor, torch.Tensor),
+            (F.permute_channels_image_pil, PIL.Image.Image),
+            (F.permute_channels_image_tensor, datapoints.Image),
+            (F.permute_channels_video, datapoints.Video),
+        ],
+    )
+    def test_dispatcher_signature(self, kernel, input_type):
+        check_dispatcher_kernel_signature_match(F.permute_channels, kernel=kernel, input_type=input_type)
+
+    def reference_image_correctness(self, image, permutation):
+        return datapoints.Image(image.numpy()[permutation, ...])
+
+    @pytest.mark.parametrize("permutation", _CORRECTNESS_PERMUTATIONS)
+    def test_image_correctness(self, permutation):
+        image = make_image()
+
+        actual = F.permute_channels(image, permutation=permutation)
+        expected = self.reference_image_correctness(image, permutation=permutation)
```

Comment:
This is testing that the indexing semantics are the same in `torch` and in `numpy`, but it doesn't test nor illustrate that the transform is working as expected. I mean, we can keep this test, but it would be good to also assert something like

```
for input_c in range(len(permutation)):
    assert_close(image[input_c, :, :], output[permutation[input_c], :, :]
```

Reply:
I've refactored the reference function to not rely on indexing, but rather "manually picking" the channels. LMK if that is sufficient.